### DISCUSSION TITLE
Is automatic Copilot suggestion possible after entering a new line (in insert mode)?

### DISCUSSION THREAD

**Discussion by tkivela** [LazyVim/LazyVim#5207] [Q&A]
My setup uses Blink and it gets Copilot suggestions on the current line ok.

However I'd like to have Copilot suggestions automatically when changing to a new line (the same way that for example VSCode does it). Is this possible?

Example:
I have a (Typescript) code comment like this:
```typescript
 // Function to calculate median of an array
```
When I'm in insert mode at the end of the comment and press enter I'm moved into a new line as expected but I don't get any completion suggestions:
![image](https://github.com/user-attachments/assets/53ec7414-d802-404d-9b49-995c0d4deeac)

In VSCode when I'm entering a new line the Copilot suggestion kicks in automatically (the multiple lines of code as ghost text):
![image](https://github.com/user-attachments/assets/59f85bfa-d2e4-4719-ae05-15fbbd25c718)
and when I press tab the suggestion is accepted
![image](https://github.com/user-attachments/assets/11400251-e502-4a40-b920-57ca55085302)

Is this possible in LazyVim with Blink?

**Comment by tkivela** [comment]
Answering to myself as I got this working, turns out that the full copilot suggestions can be turned on from the copilot plugin:
```lua
return {
  "zbirenbaum/copilot.lua",
  opts = {
    suggestion = { enabled = true },
  },
}
```

It's probably been disabled by default because blink-cmp-copilot readme (https://github.com/giuxtaposition/blink-cmp-copilot) says:
> It is recommended to disable copilot.lua's suggestion and panel modules, as they can interfere with completions properly appearing in blink-cmp-copilot.

I haven't seen any issues for myself when turning it on.

**Comment by bulletmark** [comment]
@tkivela using the current LazyVim, does this still work for you?

No matter what I do I can not get copilot suggestions for Python when opening a blank line. E.g. create the following code:

```python
# Calculate the nth number in the Fibinacci sequence
def fibinacci(n: int) -> int:
```

Using `vim` + `copilot.vim` as soon as I open the line under that `def` I get the full function implementation suggested to me by copilot. LazyVim does not, with or without `vim.g.ai_cmp = false`. BTW, your snippet above is [automatically enabled](https://www.lazyvim.org/extras/ai/copilot#copilotlua) if you set `vim.g.ai_cmp = false`.

If I create the same example in shell/bash code it also works fine in `vim` but not LazyVim.

To be clear, this problem is replicated using a completely clean fresh install of LazyVim with only `lang.python` and `ai.copilot` enabled from extras.

**Reply by folke** [reply]
This PR fixed it: https://github.com/zbirenbaum/copilot.lua/pull/318

**Reply by bulletmark** [reply]
Thanks for looking into this (and sorry for badgering you!) although as your later comment there says unfortunately that PR breaks a few things :(

**Comment by bulletmark** [comment]
Fixed with today's update of `copilot.lua`.

**Reply by webel** [reply]
Fixed as in just add
```
opts = {
    suggestion = { enabled = true },
  }
```
?

I've had no luck yet either, in my old setup I never got copilot to work with cmp and now I seem to want the opposite 😂

**Reply by bulletmark** [reply]
With a stock Lazyvim setup, the only setting you need to get this to work as I describe here (i.e. exactly the same as `copilot.vim` in vim), is to set `vim.g.ai_cmp = false` in your `~/.config/nvim/lua/config/options.lua`, and of course enable `ai.copilot` in extras.

**Reply by vlasar** [reply]
> With a stock Lazyvim setup, the only setting you need to get this to work as I describe here (i.e. exactly the same as `copilot.vim` in vim), is to set `vim.g.ai_cmp = false` in your `~/.config/nvim/lua/config/options.lua`, and of course enable `ai.copilot` in extras.

Hello @bulletmark. I'm giving this a shot. How do you accept a suggestion? As far as I can tell, [it should use the same as `blink.cmp`](https://www.lazyvim.org/extras/ai/copilot#copilotlua), but with my `blink.cmp` configuration it's just inserting a new line when the suggestion is visible.

```lua
return {
  "saghen/blink. cmp",
  opts = {
    preset = "default",
    ["<S-Tab>"] = { "select_prev", "snippet_backward", "fallback" },
    ["< Tab>"] = { "select_next", "snippet_forward", "fallback" },
    ["<CR>"] = { "accept", "fallback" },
    ["‹Esc>"] = { "hide", "fallback" },
    ...
  },
  ...
}
```

**Reply by folke** [reply]
@vlasar you're also not doing what he says?

- **remove** all your blink customizations.
- add` vim.g.ai_cmp = false` in your `options.lua`
- accept suggestion with `<tab>`

**Reply by vlasar** [reply]
> @vlasar you're also not doing what he says?
> 
>     * **remove** all your blink customizations.

Ah, right. Sorry.

I'll give it a shot, but I think I'm too used to using `<tab>` for navigation and `<cr>` for accepting.

**Reply by webel** [reply]
@vlasar you can still change the copilot keymaps to make it `<cr>` instead.

thanks @bulletmark, toggling `vim.g.ai_cmp` did the trick for me, the penny didn't drop what you meant by fixed at first.
